### PR TITLE
User preferences and Parameters

### PR DESCRIPTION
**Pull request by zizouet**
**Store user parameters in the database.**

Created 3 files : 

- **Entity** : UserPreference
- **API** : UserPreferenceAPI
- **Test** : UserPreferenceAPITest

The parameters/User_Preferences include : 
- **Theme** : Dark, Light or System
- **Text size** : Int
- **Language** : English

### CODE REVIEW

File: app/src/main/java/com/github/se/assocify/model/database/SupabaseApi.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ abstract class SupabaseApi {
         if (tag != null) {
           Log.e("API", "Error in $tag")
         }
-        Log.e("API", e.toString())
+        // Log.e("API", e.toString())
```
Comment (Mai-LinhC) [inline app/src/main/java/com/github/se/assocify/model/database/SupabaseApi.kt]: why comment this log ?

File: app/src/main/java/com/github/se/assocify/model/entities/UserPreference.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.github.se.assocify.model.entities
+
+import kotlinx.serialization.Serializable
+
+/**
+ * Represents the user preferences
+ *
+ * @param theme the theme of the application
+ * @param textSize the text size of the application
+ * @param language the language of the application
```
Comment (Mai-LinhC) [inline app/src/main/java/com/github/se/assocify/model/entities/UserPreference.kt:11]: missing userUID param

File: app/src/main/java/com/github/se/assocify/model/database/UserPreferenceAPI.kt [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  private val collectionName = "user_preference"
+
+  /**
+   * Adds a user preference to the database
+   *
+   * @param userUID the user preference to add
+   * @param onSuccess called on success
+   * @param onFailure called on failure
+   */
+  fun getUserPreference(
```
Comment (Mai-LinhC) [inline app/src/main/java/com/github/se/assocify/model/database/UserPreferenceAPI.kt]: comment not coherent with the function

File: app/src/main/java/com/github/se/assocify/model/database/UserPreferenceAPI.kt [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      onSuccess(userPreference.toUserPreference())
+    }
+  }
+
+  /**
+   * Adds a user preference to the database
+   *
+   * @param userPreference the user preference to add
+   * @param onSuccess called on success
+   * @param onFailure called on failure
```
Comment (Mai-LinhC) [inline app/src/main/java/com/github/se/assocify/model/database/UserPreferenceAPI.kt:47]: missing a parameter

File: app/src/test/java/com/github/se/assocify/model/database/UserPreferenceAPITest.kt [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+  private val userUID = "00000000-0000-0000-0000-000000000000"
+  private val userPreference =
+      UserPreference(
+          userUID = userUID, theme = Theme.DARK, textSize = 20, language = Language.ENGLISH)
+
+  @Before
+  @OptIn(ExperimentalCoroutinesApi::class)
+  fun setup() {
+    Dispatchers.setMain(UnconfinedTestDispatcher())
```
Comment (Mai-LinhC) [inline app/src/test/java/com/github/se/assocify/model/database/UserPreferenceAPITest.kt]: not needed if you use setup

File: app/src/main/java/com/github/se/assocify/model/database/SupabaseApi.kt (hunk 1 above)
Comment (zizouet) [reply app/src/main/java/com/github/se/assocify/model/database/SupabaseApi.kt]: Yes you're so right !!

File: app/src/main/java/com/github/se/assocify/model/entities/UserPreference.kt (hunk 2 above)
Comment (zizouet) [reply app/src/main/java/com/github/se/assocify/model/entities/UserPreference.kt:11]: Yes resolved thank you very much

File: app/src/main/java/com/github/se/assocify/model/database/UserPreferenceAPI.kt (hunk 3 above)
Comment (zizouet) [reply app/src/main/java/com/github/se/assocify/model/database/UserPreferenceAPI.kt]: Thank you so much hitatakimasuuuuuuuuuuu

File: app/src/test/java/com/github/se/assocify/model/database/UserPreferenceAPITest.kt (hunk 5 above)
Comment (zizouet) [reply app/src/test/java/com/github/se/assocify/model/database/UserPreferenceAPITest.kt]: That's true aligatoooo nyaaa

**Comment by Mai-LinhC** [comment]
Nitpicking about description of PR

**Review by Mai-LinhC — APPROVED** [review]
LGTM now !

**Pull request merged**